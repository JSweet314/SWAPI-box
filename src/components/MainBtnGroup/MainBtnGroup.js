import React from 'react';
import './MainBtnGroup.css';
import Button from '../Button/Button';
import PropTypes from 'prop-types';

const MainBtnGroup  = ({selectCategory, currentCategory}) => {
  const buttonTexts = ['people', 'planets', 'vehicles'];
  const buttons = buttonTexts.map((text, index) => {
    const isSelected = text === currentCategory ? 'selected': '';
    return (
      <Button 
        isSelected={isSelected}
        btnText={text} 
        key={index} 
        selectCategory={selectCategory}
      />
    );
  });

  return (
    <div className="mainBtnGroup">
      {buttons}
    </div>
  );
};

MainBtnGroup.propTypes = {
  selectCategory: PropTypes.func.isRequired,
  currentCategory: PropTypes.string.isRequired
};

export default MainBtnGroup;