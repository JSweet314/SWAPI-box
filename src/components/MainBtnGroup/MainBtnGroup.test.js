import React from 'react';
import { shallow } from 'enzyme';
import MainBtnGroup from './MainBtnGroup';

describe('MainBtnGroup', () => {
  it('should match a snapshot', () => {
    const wrapper = shallow(<MainBtnGroup />);
    expect(wrapper).toMatchSnapshot();
  });
});
