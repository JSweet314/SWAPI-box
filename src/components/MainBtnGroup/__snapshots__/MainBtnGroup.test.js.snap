// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainBtnGroup should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainBtnGroup
    currentCategory=""
    selectCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          btnText="people"
          isSelected=""
          selectCategory={[Function]}
/>,
        <Button
          btnText="planets"
          isSelected=""
          selectCategory={[Function]}
/>,
        <Button
          btnText="vehicles"
          isSelected=""
          selectCategory={[Function]}
/>,
      ],
      "className": "mainBtnGroup",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "btnText": "people",
          "isSelected": "",
          "selectCategory": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "btnText": "planets",
          "isSelected": "",
          "selectCategory": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "btnText": "vehicles",
          "isSelected": "",
          "selectCategory": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            btnText="people"
            isSelected=""
            selectCategory={[Function]}
/>,
          <Button
            btnText="planets"
            isSelected=""
            selectCategory={[Function]}
/>,
          <Button
            btnText="vehicles"
            isSelected=""
            selectCategory={[Function]}
/>,
        ],
        "className": "mainBtnGroup",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "btnText": "people",
            "isSelected": "",
            "selectCategory": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "btnText": "planets",
            "isSelected": "",
            "selectCategory": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "btnText": "vehicles",
            "isSelected": "",
            "selectCategory": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
